modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/util/v1_0/ShipmentItemUtil.java

```diff
@@ -52,16 +59,91 @@ public static CommerceShipmentItem addOrUpdateShipmentItem(
 				commerceShipmentItem.getCommerceInventoryWarehouseId();
 		}
 
+		CommerceInventoryWarehouse commerceInventoryWarehouse =
+			getCommerceInventoryWarehouse(
+				serviceContext.getCompanyId(), shipmentItem,
+				commerceInventoryWarehouseService);
+
+		CommerceOrderItem commerceOrderItem = getCommerceOrderItem(
+			serviceContext.getCompanyId(), shipmentItem,
+			commerceOrderItemService);
+
 		return commerceShipmentItemService.addOrUpdateCommerceShipmentItem(
 			externalReferenceCode, commerceShipment.getCommerceShipmentId(),
 			GetterUtil.getLong(
-				shipmentItem.getOrderItemId(), defaultOrderItemId),
+				commerceOrderItem.getCommerceOrderItemId(), defaultOrderItemId),
 			GetterUtil.getLong(
-				shipmentItem.getWarehouseId(), defaultWarehouseId),
+				commerceInventoryWarehouse.getCommerceInventoryWarehouseId(),
+				defaultWarehouseId),
```

Comment:
because there are defaults I think it's wrong that inside the new getCommerceOrderItem or getCommerceInventoryWarehouse we throws exception if we not have the data. I think we need to pass the default to these methods and return them when we are not able to find other data